packages/sdk/src/client.ts

```diff
@@ -664,6 +669,48 @@ export function SoundClient({
     return mintSchedules.flat().sort((a, b) => a.startTime - b.startTime)
   }
 
+  async function getContractError(txHash: string) {
+    if (
+      txHash === NULL_BYTES32 ||
+      txHash.slice(0, 2) !== '0x' ||
+      // Tx hash is 32 bytes, which is 64 hex characters + '0x'
+      txHash.length !== 66
+    ) {
+      throw new InvalidTxHashError({ txHash })
+    }
+
+    const provider = await _requireProvider()
+    const tx = await provider.getTransaction(txHash)
+
+    try {
+      // Simulate the original transaction
+      const response = await provider.call(
+        {
+          to: tx.to,
+          from: tx.from,
+          nonce: tx.nonce,
+          gasLimit: tx.gasLimit,
+          gasPrice: tx.gasPrice,
+          data: tx.data,
+          value: tx.value,
+          chainId: tx.chainId,
+          type: tx.type ?? undefined,
+          accessList: tx.accessList,
+        },
+        tx.blockNumber,
+      )
```

Comment:
The only way to know why a transaction failed after it was mined is to simulate the original call at the block it was mined on. The reason is because nothing is stored or logged about the error on-chain unless specifically implemented in the contract.

Reply:
Is this something more general that should be in webapp?

Seems useful for deploy/withdraw type transactions as well

Reply:
Main benefit of it being in SDK is anyone building on the protocol can use it. 

Imo, if we end up needing it in webapp anywhere except minting, its probably a sign we're lacking good code & testing earlier in the code paths, since the other txs aren't prone to failing due to rapid on-chain storage changes like mint txs. 

We could also check if those transactions will fail ahead of time [by using `callStatic`](https://linear.app/sound/issue/ENG-1994/call-callstatic-before-calling-contract-methods)

That said, we should prob eventually add all the contract errors so independent builders can debug any failures.

Reply:
Added the remaining errors for completeness but don't have tests for them since this isn't a super critical helper. https://github.com/soundxyz/sdk/pull/175/commits/88f5b3d5e442f153e76a4eb1eb7b41e40666d9df